internal/wire/op_msg_test.go

```diff
@@ -147,6 +148,115 @@ var msgTestCases = []testCase{{
 	name:      "msg_fuzz1",
 	expectedB: testutil.MustParseDumpFile("testdata", "msg_fuzz1.hex"),
 	err:       `wire.OpMsg.readFrom: invalid kind 1 section length -13619152`,
+}, {
+	name: "NaN",
+	expectedB: []byte{
+		0x79, 0x00, 0x00, 0x00, // MessageLength
+		0x11, 0x00, 0x00, 0x00, // RequestID
+		0x00, 0x00, 0x00, 0x00, // ResponseTo
+		0xdd, 0x07, 0x00, 0x00, // OpCode
+		0x00, 0x00, 0x00, 0x00, // FlagBits
+		0x00,                   // section 0 kind
+		0x64, 0x00, 0x00, 0x00, // document size
+		0x02, 0x69, 0x6e, 0x73, 0x65, 0x72, 0x74, 0x00, // string "insert"
+		0x07, 0x00, 0x00, 0x00, // "values" length
+		0x76, 0x61, 0x6c, 0x75, 0x65, 0x73, 0x00, // "values"
+		0x04, 0x64, 0x6f, 0x63, 0x75, 0x6d, 0x65, 0x6e, 0x74, 0x73, 0x00, // array "documents"
+		0x29, 0x00, 0x00, 0x00, // document (array) size
+		0x03, 0x30, 0x00, // element 0 (document)
+		0x21, 0x00, 0x00, 0x00, // element 0 size
+		0x01, 0x76, 0x00, // double "v"
+		0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf8, 0x7f, // NaN
+		0x07, 0x5f, 0x69, 0x64, 0x00, // ObjectID "_id"
+		0x63, 0x77, 0xf2, 0x13, 0x75, 0x7c, 0x0b, 0xab, 0xde, 0xbc, 0x2f, 0x6a, // ObjectID value
+		0x00,                                                       // end of element 0 (document)
+		0x00,                                                       // end of document (array)
+		0x08, 0x6f, 0x72, 0x64, 0x65, 0x72, 0x65, 0x64, 0x00, 0x01, // "ordered" true
+		0x02, 0x24, 0x64, 0x62, 0x00, // "$db"
+		0x05, 0x00, 0x00, 0x00, // "test" length
+		0x74, 0x65, 0x73, 0x74, 0x00, // "test"
```

Comment:
All that is not really needed since the only thing we need is the simplest document: `{"": NaN}`

Reply:
The simplest document won't find the lack of recursion inside the validation function though.